string name
int32 channel
bool value
float64 avalue
time before
